### ISSUE TITLE
Intent vs. Benefits and conformance (with SC 1.2.3 and SC 1.2.5)

### ISSUE THREAD

**Issue by brennanyoung** [w3c/wcag#1042]
**Background**
I've been struggling for months to understand whether a special variant of Advisory technique H96 (where the audio description cues are fed into an aria-live region for AT announcement) complies with or violates SC 1.2.3 and/or SC 1.2.5. 

All the a11y professionals I have consulted seem to prefer to tie themselves in knots or quote vaguely relevant WCAG copy rather than give me a straight answer to this apparently simple question. I still don't have a clear answer.

So I reformulated the question on Slack in a deliberately foolish way: "I assume that users needing audio descriptions will also be using ATs such as screen readers. Can anyone confirm or challenge this assumption?"

This produced many, many answers, lots of finger-wagging and (finally) some clarity: "Many, maybe most, people using screen reader benefit from Audio Descriptions, but not everyone who uses Audio Descriptions is a screen reader user." and some vague comments about cognitive impairment. None of this is obvious. And I still have some doubts.

When I look at the relevant "Understanding..." documents about these success criteria, I am told that the intent is to serve people who are **blind or visually impaired**. In that case, our H96 variant should be compliant. Cognitive impairment is **not** mentioned under "intent" but it does get mentioned under "benefits". It's not at all clear whether serving the (unspecified) needs of cognitively impaired users is required to be compliant here.

**Intent vs. Benefits, when they contradict each other**
This is the main issue I wish to raise.

If the "Intent" section of an "Understanding" document mentions users of group A, but does not mention the users of group B mentioned in the "Benefits", does this mean that meeting the needs of both groups A, and B is required to be compliant? Or is it enough to meet the needs of group A. This is very unclear.

I suggest that these particular docs need refinement on these points, or if the apparent contradiction is **intentional**, perhaps the "Understanding Conformance" doc could clarify the relative importance of "Intent" and "Benefits" when judging whether a success criterion is violated.

**Comment by patrickhlauke** [comment]
just to pile some more vague handwaving on this (as yes, in many cases, by their very nature, it IS hard to give a definitive PASS/FAIL on very specific implementation attempts)

> When I look at the relevant "Understanding..." documents about these success criteria, I am told that the intent is to serve people who are blind or visually impaired. 

not all visually impaired users use a screen reader. some may use a screen magnifier, which may make it hard for them to be able to see the entirety of a video due to their reduced viewport/field of view. and while some screen magnifiers also include screen reading/voicing, that's not always the case.

**Comment by brennanyoung** [comment]
@patrickhlauke  that's **very** useful information, thank you. Learning all the time. It seems too obvious to mention, and yet it wasn't obvious enough for me to imagine, or for anyone but you to mention it until now! I therefore think it would be valuable to mention the use case of Audio Descriptions without a screen reader (or other announcer AT) on H96, (and/or perhaps on understanding 1.2.3/1.2.5).

Additionally, I think it would be valuable to consider that Audio Descriptions for blind/partially sighted users may need to be contain qualitatively different cues than Audio Descriptions for those with cognitive impairments. [This recent research](http://epubs.surrey.ac.uk/848660/) explores that very area. I would appreciate if WCAG could provide some guidance about whether and how to distinguish these two kinds (actually two quite separate use cases) of Audio Description.

Incidentally I'm also preparing (what I hope will be) a sufficient technique based on H96, with a switch to select the **Web Speech API** instead of an aria-live region for audio descriptions. This would (I believe) handle all the aforementioned use cases without requiring prerecorded audio tracks.

But still outstanding (and the main topic of this thread) is the need to handle the potential for confusion about compliance when an "Understanding" document has this form:
 
"Intent: Serves the needs of group A"
and
"Benefits: Group A and B benefit from this."

(where groups A and B represent different kinds of disabled user)

**Comment by alastc** [comment]
It's a tricky case, I'll approach this from a 'first principles' point of view 

These can both be true, and quite often are:
1. The intent, the reason for the success criteria (SC) to exist, is to serve group A. 
2. Following the SC has benefits for group A and (secondarily) can have benefits for group B.

Where you have an 'also benefits' situation it might have less basis than for group A, or might not be as important for group B.

I think the more important aspect of your question is how success criteria (SC) & techniques relate:
- A sufficient technique is a possible way to meet the SC. It may go beyond meeting the SC, but it is a known way to at least meet it.
- An advisory technique may not meet the SC, but there is some benefit to using it.

H96 is advisory because it may not meet the SC due to support from AT. There may be some benefit, but it is not sufficient on its own.


> [does]  a special variant of Advisory technique H96 (where the audio description cues are fed into an aria-live region for AT announcement) complies with or violates SC 1.2.3 and/or SC 1.2.5.

1.2.3/5 require that "An ... audio description of the pre-recorded video content is provided for synchronized media". 

So yes, it is provided. _But_ then it is requiring the user to have AT (a screenreader) to access it. In this sort of case you then have to consider whether it is [accessibility supported](https://www.w3.org/TR/WCAG21/#dfn-accessibility-supported). This means that the solution must “have accessibility-supported user agents that are available to users”.

Given that not everyone with a visual impairment would use a screenreader, and I don’t know of a widely available plugin that does an equivalent, I would overall say:
No, this wouldn’t meet 1.2.3/5. (Other opinions are avialable.)

Taking a step back, I’d note that the timings on ARIA-live are not consistent (or required to be consistent) across screenreaders, so you will probably find it does not work in practice.

If I were tasked with something like this, I’d probably look at a web-audio solution and put a button on the interface for everyone… that would meet the SC.

**Comment by alastc** [comment]
Hi @brennanyoung, does that answer your question? If so I can close this issue, thanks.

**Comment by brennanyoung** [comment]
My concrete implementation question has been answered. 
Yet, I would still welcome some brief mention on the H96 page of the specific, not-at-all-obvious shortcomings of relying on live regions to communicate the AD cues.

**Comment by alastc** [comment]
H96 wouldn't be the place to highlight shortcomings with a completely separate technique. The "H" is for HTML, anything using ARIA would be under the ARIA techniques. Except that we don't want to encourage that approach so we wouldn't add one!

If I had to put that somewhere, I think it would be best in the understanding documents for the audio-desc related SCs.

It is unusual for us to add a seemingly random "please don't do this" comment though, the documents are long enough just saying what people should do. Also including every possible thing they shouldn't do adds to the complexity.

**Comment by brennanyoung** [comment]
Perhaps something like "if you rely only on text tracks to supply AD cues, make sure that the implementation does not require a given class of AT (such as a screen reader) to communicate those cues to the user". No "don't do this" and no mention of ARIA.

**Comment by alastc** [comment]
Pinging @johnfoliot, I think you might remember the discussion about [H96](https://www.w3.org/WAI/WCAG21/Techniques/html/H96) and JS polyfills?

Back in Feb 2019 we added this at the end of the description:
> As of February 2019 when this Advisory technique was last reviewed by the Working Group, there is no native support in user agents for this technique. However, support is available via JavaScript polyfills. 

The issue above is about creating an audio description by feeding an aria-live region for the AT to announce. That doesn't seem like a useful approach given the lack of predictable timing for aria-live.

Could we add something to the warning? E.g. 
> ... However, support is available via JavaScript polyfills. The use of JavaScript to update an aria-live region is not recommended due to the unpredictable timing.

**Comment by OwenEdwards** [comment]
Reading through this thread, I want to note that I've done quite a lot of work on this with the [video.js](https://videojs.com/) player framework, following on from some work we did at the [Video Description R&D Center](http://www.vdrdc.org/).

There are a lot of limitations to using a text track in an `aria-live` region for audio description, including, as @alastc mentions, that it _only_ works if the user has an AT which voices `aria-live` content. (There are also issues with things like random access/seeking, where a user might land in the middle of the time allotted for a specific text cue, but the AT would be presented with _all_ of the text of that cue, and therefore may over-run the time allotted for the cue when it speaks, and "step on" other dialog).

That was part of the reason that I built out this demo which, rather than using ARIA and relying on AT, uses purely HTML/CSS/JavaScript and relies on browser text-to-speech (the `SpeechSynthesis` part of the [Web Speech API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Speech_API)):

http://www.ca11y.com/videojs-speak-descriptions-track/

Note that this demo is definitely **just a prototype**, and is **not intended for production systems**. But what it does do, which `aria-live` and an AT cannot, is that if the time to speak the text of a cue is going to take too long for the amount of time on the video timeline available for that cue (since every cue in a text track has a _start_ and _end_ time on the video timeline), it pauses the video until the text-to-speech engine finishes, and then re-starts the video. This is somewhat similar to "extended description" (SC 1.2.7), but it's actually a hybrid of inline and extended description which is intended to encourage more discussion around what complying with SC 1.2.5 _could_ look like when using a text track and synthesized speech.

Personally, I think this is a much more workable solution to the idea of using a text track to provide audio description than H96. And I think it's very relevant to your question, @brennanyoung. I don't want to derail this thread from your original question, but I think it adds to the discussion - let me know what you think.
